### DISCUSSION TITLE
Convert between panes and tabs

### DISCUSSION THREAD

**Discussion by cryocz** [zellij-org/zellij#2397] [Q&A] [answered]
Is there a way to convert a pane into a tab and/or vice versa?

**Comment by Firehawke** [comment]
Popping in here to add that this is a question I'd like to know, myself.

Would be nice if I could convert e.g. my TUI email client between a floating pane and a full-screen tab depending on what I need.

**Reply by Lockszmith-GH** [reply]
Back in 2023, when OP asked, it wasn't possible.

The features exist today:
- [Break Pane](https://zellij.dev/documentation/keybindings-possible-actions.html#breakpane) (Default keyboard binding is <kbd>Ctrl</kbd>+<kbd>t</kbd> and then <kbd>b</kbd>/<kbd>[</kbd>/<kbd>]</kbd>)
- [TogglePaneEmbedOrFloating](https://zellij.dev/documentation/keybindings-possible-actions.html#togglepaneembedorfloating) (Default keyboard binding is <kbd>Ctrl</kbd>+<kbd>p</kbd> and then <kbd>e</kbd>)

**Comment by ShreeChaturvedi** [comment, marked as answer]
One key correction: `BreakPane` lives in tab mode, not pane mode, so it's `Ctrl+t` then `b`. `Ctrl+p` `b` isn't bound to anything.

From `default.kdl`:

```kdl
tab {
    bind "b" { BreakPane; SwitchToMode "Normal"; }        // pane out into its own tab
    bind "]" { BreakPaneRight; SwitchToMode "Normal"; }   // send it into the next tab
    bind "[" { BreakPaneLeft; SwitchToMode "Normal"; }    // send it into the previous tab
}
pane {
    bind "e" { TogglePaneEmbedOrFloating; SwitchToMode "Normal"; }
}
```

`Ctrl+t` `[` or `]` is the tab-back-into-a-pane direction, and `Ctrl+p` `e` is the floating/embedded flip @Firehawke wants for the mail client.

Only that last one has a CLI equivalent (`zellij action toggle-pane-embed-or-floating`). The break actions are keybinding-only as of 0.44.0, so you can't script them.

**Reply by Lockszmith-GH** [reply]
Thanks @ShreeChaturvedi - fixed it.
